### PR TITLE
Reconcile flake and non-flake builds.

### PR DESCRIPTION
**Pull request by shlevy**
Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

**Comment by shlevy** [comment]
Note that I haven't yet tested the new github action, will likely do so after merge.

**Comment by shlevy** [comment]
@michaelpj I don't have ability to request reviewers but I think you'd be best.

File: nix/sources.nix
Comment (michaelpj) [inline nix/sources.nix:18]: Worth a comment about why we use the builtin fetcher for nixpkgs but not for the rest.

File: nix/sources.nix
Comment (michaelpj) [reply nix/sources.nix:3]: I wonder if this is something we could upstream to `flake-utils`? Seems useful.

**Review by michaelpj — APPROVED** [review]

File: .github/workflows/flakes-update.yml
Comment (shlevy) [reply .github/workflows/flakes-update.yml:6]: @michaelpj Re timing, this was the same time as the old job, not sure yet how this one will behave when there's no change. Can make it less often either way.

Re blacklist: Those dependencies are pinned to a specific revision in flake.nix so the update won't touch them.

File: flake.nix
Comment (shlevy) [reply flake.nix:16]: @michaelpj They do, but I was getting an error when I specified both a revision and a branch. Specifying the revision here was so that we don't automatically bump nixpkgs.

File: flake.nix
Comment (shlevy) [reply flake.nix:22]: It's for pinning the packages we don't want to auto-update.

File: nix/default.nix
Comment (shlevy) [reply nix/default.nix:32]: The version of nixpkgs we're using had a code path that uses `builtins.currentSystem` if we don't pass `system` this way, which the flake eval complains about.

File: nix/sources.nix
Comment (shlevy) [reply nix/sources.nix:1]: @michaelpj This matches the interface the autogenerated `sources.nix` had but you're right that there's no real need. Will change.

File: nix/sources.nix
Comment (shlevy) [reply nix/sources.nix:3]: :+1:

File: nix/sources.nix
Comment (shlevy) [reply nix/sources.nix:6]: :+1:

File: flake.nix
Comment (shlevy) [reply flake.nix]: Fixed.

File: flake.nix
Comment (shlevy) [reply flake.nix]: Fixed.

File: flake.nix
Comment (michaelpj) [reply flake.nix:29]: Sounds like it needs a comment!

File: nix/default.nix
Comment (shlevy) [reply nix/default.nix:32]: Will add a comment.

File: flake.nix
Comment (shlevy) [reply flake.nix:29]: :+1:

File: nix/sources.nix
Comment (shlevy) [reply nix/sources.nix]: Fixed

File: nix/sources.nix
Comment (shlevy) [reply nix/sources.nix]: Fixed

File: nix/sources.nix
Comment (shlevy) [reply nix/sources.nix]: Fixed

File: nix/default.nix
Comment (michaelpj) [reply nix/default.nix:32]: Okay I'm just confused about the difference between passing `system` and `localSystem`!

File: nix/sources.nix
Comment (shlevy) [reply nix/sources.nix:4]: Fixed

File: flake.nix
Comment (shlevy) [reply flake.nix:31]: Fixed.

File: nix/default.nix
Comment (shlevy) [reply nix/default.nix:35]: Fixed.

**Comment by shlevy** [comment]
@michaelpj Comments addressed, let me know what you want to do about the cron job.

**Comment by michaelpj** [comment]
Happy to merge the cron job if you're willing to fix if it becomes a pain :p

**Review by michaelpj — APPROVED** [review]

**Pull request merged**
